File: welcome/welcome.py
# API

class IO:
	@staticmethod
	def print(*arguments):
		args = ""
		for arg in arguments:
			args += str(arg)
		print(args)

class Fiber:
	def __init__(self, fn):
		self.gen = None
		self.fn = fn
		self.done = False

	def call(self):
		try:
			if self.gen is None:
				ret = self.fn()
				self.gen = ret;
				return next(self.gen)
			else:
				return next(self.gen)
		except StopIteration:
			self.done = True
		return None

	@property
	def isDone(self):
		return self.done

# script

IO.print("hello, world!");

def unnamed_lambda0():
	for word in ["small", "clean", "fast"]:
		yield word


adjectives = Fiber(unnamed_lambda0);

while not adjectives.isDone:
	IO.print(adjectives.call())

File: welcome/welcome.js
'use strict';
// API

class IO {
	static print() {
		var args = "";
		for(var arg of arguments) {
			args += arg;
		}
		console.log(args);
	}
}

class Fiber {
	constructor(fn) {
		this.generator = fn();
		this.done = false;
	}
	
	call() {
		let v = this.generator.next();
		this.done = v.done;
		return v.value;
	}

	get isDone() {
		return this.done;
	}
}

// script

IO.print("hello, world!");

let adjectives = new Fiber(function*() {
	yield * ["small", "clean", "fast"][Symbol.iterator](function*(word) {
		yield word;
	});
});

while(!adjectives.isDone) IO.print(adjectives.call());
